### ISSUE TITLE
collect parameter commit broke metric counter(s)?

### ISSUE THREAD

**Issue by frimik** [prometheus/mysqld_exporter#245]
I was basing another prometheus exporter on this and noticed that things weren't working as intended for me ... I went back in history a bit and found that 8b92fe8a393d3f3d48b2f3bad2195d68ed8a893f broke at least one global metric counter: 

```# HELP mysql_exporter_scrapes_total Total number of times MySQL was scraped for metrics.
# TYPE mysql_exporter_scrapes_total counter
mysql_exporter_scrapes_total 2
```

On HEAD it's forever stuck on 2 (it gets init and counts to 2 on every scrape).

Sorry, edit; it's #235 that broke this. Messed up my history a bit.

**Comment by frimik** [comment]
Sorry, it's #235 that broke this. Messed up my history a bit.

**Comment by SuperQ** [comment]
Fixed with #292.
